graphistry/plugins/spannergraph.py

```diff
@@ -0,0 +1,294 @@
+import os
+import pandas as pd
+import json
+import time
+from typing import Any, List, Dict
+
+from graphistry.util import setup_logger
+logger = setup_logger(__name__)
+
+import logging
+logging.basicConfig(level=logging.INFO)
+
+from google.cloud.spanner_v1.data_types import JsonObject
+
+class SpannerConnectionError(Exception):
+    """Custom exception for errors related to Spanner connection."""
+    pass
+
+class SpannerQueryResult:
+    """
+    Encapsulates the results of a query, including metadata.
+
+    :ivar list data: The raw query results.
+    :ivar float execution_time: The time taken to execute the query.
+    :ivar int record_count: The number of records returned.
+    """
+
+    def __init__(self, data: List[Any], execution_time: float):
```

Comment:
I'm guessing we can't get any more granuarly nested typing, like `List[Dict[str,...` or `List[Union[Kind1, Kind2, ...]]` ?